RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -305,214 +498,207 @@ void EvtPlaneProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSetup
         }
       }
     }
-
-    //
-    //Get flattening parameter file.
-    //
-    if (loadDB_) {
-      edm::ESHandle<RPFlatParams> flatparmsDB_;
-      iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
-      LoadEPDB db(flatparmsDB_, flat);
-      if (!db.IsSuccess()) {
-        loadDB_ = kFALSE;
-      }
+  }
+  //
+  //Get flattening parameter file.
+  //
+  if (loadDB_ && hirpWatcher.check(iSetup)) {
+    edm::ESHandle<RPFlatParams> flatparmsDB_;
+    iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
+    LoadEPDB db(flatparmsDB_, flat);
+    if (!db.IsSuccess()) {
+      loadDB_ = kFALSE;
     }
-
-  }  //rp record change
-
+  }
   //
   //Get Centrality
   //
   int bin = 0;
+  int cbin = 0;
   if (loadDB_) {
     edm::Handle<int> cbin_;
     iEvent.getByToken(centralityBinToken, cbin_);
-    int cbin = *cbin_;
+    cbin = *cbin_;
     bin = cbin / CentBinCompression_;
   }
   //
   //Get Vertex
   //
-  int vs_sell = 0.;
-  float vzr_sell;
-  iEvent.getByToken(vertexToken, vertex_);
-  const reco::VertexCollection *vertices3 = nullptr;
-  if (vertex_.isValid()) {
-    vertices3 = vertex_.product();
-    vs_sell = vertices3->size();
-  }
-  if (vs_sell > 0) {
-    vzr_sell = vertices3->begin()->z();
-  } else
-    vzr_sell = -999.9;
-  //
-  for (int i = 0; i < NumEPNames; i++)
-    rp[i]->reset();
-  if (vzr_sell < minvtx_ or vzr_sell > maxvtx_)
+  edm::Handle<reco::VertexCollection> vertices;
+  iEvent.getByToken(vertexToken, vertices);
+
+  //best vertex
+  const reco::Vertex &vtx = (*vertices)[0];
+  double bestvz = vtx.z();
+  double bestvx = vtx.x();
+  double bestvy = vtx.y();
+  double bestvzError = vtx.zError();
+  double bestvxError = vtx.xError();
+  double bestvyError = vtx.yError();
+  math::XYZPoint bestvtx(bestvx, bestvy, bestvz);
+  if (bestvz < minvtx_ || bestvz > maxvtx_)
     return;
 
-  //calorimetry part
-
-  double tower_eta, tower_phi;
-  double tower_energyet, tower_energyet_e, tower_energyet_h;
-
-  iEvent.getByToken(caloToken, caloCollection_);
+  for (int i = 0; i < NumEPNames; i++)
+    rp[i]->reset();
+  edm::Handle<edm::ValueMap<float>> chi2Map;
+  edm::Handle<edm::View<pat::PackedCandidate>> cands;
+  edm::Handle<reco::PFCandidateCollection> calocands;
+  if (strack.find("packedPFCandidates") != std::string::npos) {
```

Comment:
compute a boolean flag in the constructor and use in event by event, no need to check the string repeatedly